lib/fixup.rb

```diff
@@ -21,8 +28,28 @@ def require_gem(gem_name_and_require = nil, require_override = nil)
 end
 require_gem('highline/import')
 
+# All needed constants & variables should be defined here in case they
+# have not yet been loaded or or fixup is being run standalone.
+
 CREW_VERBOSE = ARGV.intersect?(%w[-v --verbose]) unless defined?(CREW_VERBOSE)
 
+HOME = '/home/chronos/user' unless defined?(HOME)
+CREW_PREFIX = '/usr/local' unless defined?(CREW_PREFIX)
+CREW_LIB_PATH = File.join(CREW_PREFIX, 'lib/crew') unless defined?(CREW_LIB_PATH)
+CREW_CONFIG_PATH = File.join(CREW_PREFIX, 'etc/crew') unless defined?(CREW_CONFIG_PATH)
+CREW_META_PATH = File.join(CREW_CONFIG_PATH, 'meta') unless defined?(CREW_CONFIG_PATH)
+# via git log --reverse --oneline lib/const.rb | head -n 1
+CREW_CONST_GIT_COMMIT = '72d807aac' unless defined?(CREW_CONST_GIT_COMMIT)
+CREW_REPO   = 'https://github.com/chromebrew/chromebrew.git' unless defined?(CREW_REPO)
+CREW_BRANCH = 'master' unless defined?(CREW_BRANCH)
+unless defined?(ARCH)
+  KERN_ARCH = Etc.uname[:machine] unless defined? KERN_ARCH
+  ARCH = %w[aarch64 armv8l].include?(KERN_ARCH) ? 'armv7l' : KERN_ARCH
+end
+LIBC_VERSION = Etc.confstr(Etc::CS_GNU_LIBC_VERSION).split.last unless defined?(LIBC_VERSION)
+CREW_PACKAGES_PATH = File.join(CREW_LIB_PATH, 'packages') unless defined?(CREW_PACKAGES_PATH)
+@device = PackageUtils.load_json unless defined? @device
```

Comment:
Wouldn't it be easier just to `require 'const'`?

Reply:
For sure. This is an exercise in throwing out dependencies to see what happens. This was useful in tracking down an issue in `const.rb`, so maybe that's useful?

Reply:
Why don't we have tests for const.rb instead?

Reply:
This file gets loaded on crew update via git pull, and thus there may be new constants in const.rb that don't exist yet because the const.rb that was loaded is older, so this is really just a superset of the prior approach of checking for the define of a constant before it is first used in `fixup.rb`.